api/src/main/java/com/datatorrent/api/Context.java

```diff
@@ -301,6 +301,12 @@
      */
     int getId();
 
+    /**
+     * Return the number of windows before the next checkpoint including the current window.
+     * @return Number of windows from checkpoint, 1 if the checkpoint will be after the current window
+     */
+    int getWindowsFromCheckpoint();
```

Comment:
@PramodSSImmaneni If an operator is inside a widow and checkpoint will happen after that, should this not return 0? 1 means that there is still one window before checkpoint will happen which is not the case right